<a href="https://colab.research.google.com/github/jakecho1108/Project1/blob/main/Project_1_Part_2_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Data Science Sales Insights

Jake Cho

##Imports

In [4]:
import pandas as pd

## Loading Data

In [5]:
df = pd.read_csv('/content/sales_predictions.csv')
df.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

##Data Cleaning

In [6]:
df.shape

(8523, 12)

There are 12 columns and 8,523 rows.

In [7]:
df.dtypes

Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
dtype: object

#Duplicated Data

In [8]:
df.duplicated().sum()

0

There are no duplicated data

#Missing Values

In [9]:
df.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

In [10]:
df.isna().sum()/len(df)*100

Item_Identifier               0.000000
Item_Weight                  17.165317
Item_Fat_Content              0.000000
Item_Visibility               0.000000
Item_Type                     0.000000
Item_MRP                      0.000000
Outlet_Identifier             0.000000
Outlet_Establishment_Year     0.000000
Outlet_Size                  28.276428
Outlet_Location_Type          0.000000
Outlet_Type                   0.000000
Item_Outlet_Sales             0.000000
dtype: float64

There are 1463 missing values in (17.2% of data points) in Item Weight column.

There are 2410 missing values in (28.3% of data points) in Outlet Size column.

In [11]:
df.dropna(axis=1, inplace=True) 
df

Item_Identifier Item_Fat_Content  Item_Visibility              Item_Type  \
0              FDA15          Low Fat         0.016047                  Dairy   
1              DRC01          Regular         0.019278            Soft Drinks   
2              FDN15          Low Fat         0.016760                   Meat   
3              FDX07          Regular         0.000000  Fruits and Vegetables   
4              NCD19          Low Fat         0.000000              Household   
...              ...              ...              ...                    ...   
8518           FDF22          Low Fat         0.056783            Snack Foods   
8519           FDS36          Regular         0.046982           Baking Goods   
8520           NCJ29          Low Fat         0.035186     Health and Hygiene   
8521           FDN46          Regular         0.145221            Snack Foods   
8522           DRG01          Low Fat         0.044878            Soft Drinks   

      Item_MRP Outlet_Identifier  Outlet_Establishment_Year  \
0     249.8092            OUT049                       1999   
1      48.2692            OUT018                       2009   
2     141.6180            OUT049                       1999   
3     182.0950            OUT010                       1998   
4      53.8614            OUT013                       1987   
...        ...               ...                        ...   
8518  214.5218            OUT013                       1987   
8519  108.1570            OUT045                       2002   
8520   85.1224            OUT035                       2004   
8521  103.1332            OUT018                       2009   
8522   75.4670            OUT046                       1997   

     Outlet_Location_Type        Outlet_Type  Item_Outlet_Sales  
0                  Tier 1  Supermarket Type1          3735.1380  
1                  Tier 3  Supermarket Type2           443.4228  
2                  Tier 1  Supermarket Type1          2097.2700  
3                  Tier 3      Grocery Store           732.3800  
4                  Tier 3  Supermarket Type1           994.7052  
...                   ...                ...                ...  
8518               Tier 3  Supermarket Type1          2778.3834  
8519               Tier 2  Supermarket Type1           549.2850  
8520               Tier 2  Supermarket Type1          1193.1136  
8521               Tier 3  Supermarket Type2          1845.5976  
8522               Tier 1  Supermarket Type1           765.6700  

[8523 rows x 10 columns]

The reason why I dropped two columns rather than the rows is because I believed that for the sake of the project, it's better to have more datapoints with two less columns than get rid of 40% of the current dataset. Also, the size of the outlet and the weight of the item aren't the most crucial information to keep.

In [12]:
df.isna().sum()

Item_Identifier              0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

There are no missing values after filtering them out.

In [19]:
df['Item_Fat_Content'].value_counts()
df['Item_Fat_Content']=df['Item_Fat_Content'].replace('LF','Low Fat')
df['Item_Fat_Content']=df['Item_Fat_Content'].replace('low fat','Low Fat')
df['Item_Fat_Content']=df['Item_Fat_Content'].replace('reg','Regular')
df['Item_Fat_Content'].value_counts()

Low Fat    5517
Regular    3006
Name: Item_Fat_Content, dtype: int64

Only Item_Fat_Content had inconsistent categorization.

In [31]:
df['Item_Outlet_Sales'].value_counts()

958.7520     17
1342.2528    16
703.0848     15
1845.5976    15
1278.3360    14
             ..
4124.6310     1
6622.7126     1
1614.5650     1
5602.7070     1
2778.3834     1
Name: Item_Outlet_Sales, Length: 3493, dtype: int64

In [44]:
df[['Item_Visibility', 'Item_MRP', 'Outlet_Establishment_Year', 'Item_Outlet_Sales']].min()

Item_Visibility                 0.00
Item_MRP                       31.29
Outlet_Establishment_Year    1985.00
Item_Outlet_Sales              33.29
dtype: float64

In [45]:
df[['Item_Visibility', 'Item_MRP', 'Outlet_Establishment_Year', 'Item_Outlet_Sales']].mean()


Item_Visibility                 0.066132
Item_MRP                      140.992782
Outlet_Establishment_Year    1997.831867
Item_Outlet_Sales            2181.288914
dtype: float64

In [46]:
df[['Item_Visibility', 'Item_MRP', 'Outlet_Establishment_Year', 'Item_Outlet_Sales']].max()

Item_Visibility                  0.328391
Item_MRP                       266.888400
Outlet_Establishment_Year     2009.000000
Item_Outlet_Sales            13086.964800
dtype: float64